client/src/main/scala/org/http4s/client/PoolManager.scala

```diff
@@ -171,14 +215,16 @@ private final class PoolManager[F[_], A <: Connection[F]](
               s"Connection returned could not be recycled, new connection needed: $stats")
             val Waiting(key, callback) = waitQueue.dequeue()
```

Comment:
Same problem as line 202: we don't know that the first waiter in the queue is only blocked on `maxTotal`.

Reply:
It will just be queued back if limit allows, right? We can just keep on dequeuing until something succeeds, no?

Reply:
I didn't notice that the `createConnection` method changed, sorry. I still think this is not what we want since it will effectively deprioritize a request if it gets put back at the end of the `waitQueue` and doesn't try another waiter. I kind of liked it the way it was, as a verbose assert.

Reply:
@bryce-anderson @rossabaker let me know what you think the behaviour should be. Do we put it on waitQueue or what?

Reply:
We want to pick a waiter with something like
```scala
waitQueue.dequeueFirst(numConnectionsCheckHolds(_))
```
It's arguable that if we don't find one that we shouldn't have kill the connection, but that makes the `dequeueFirst` predicate different since the maxConn part will fail.

Reply:
And I think this is like the other discussion, except we don't put the connection back on the idle queue if we can't find a waiter.